io_scene_godot/__init__.py

```diff
@@ -112,6 +112,13 @@ class ExportGodot(bpy.types.Operator, ExportHelper):
         description="Export only objects on the active layers.",
         default=True,
         )
+    use_material_transparent = EnumProperty(
+        name='Export Material As Transparent',
```

Comment:
I'm not convinced this should be a scene/export parameter. Can we not infer this from the node tree somehow (existence of a transparant BSDF perhaps?)

Reply:
This configuration is used to control whether the `color.a` (every bsdf output has color.a) is written to `ALPHA` in godot shader. Godot  engine treat every material written to `ALPHA` to go through the transparent pipeline, it means object needed to be rendered from back to front, which can not take advantage of depth map and is inefficient.    

also I find even set `ALPHA` = 1.0, the material still seems transparent, I suppose `blend_mode` and `cull_back_face` need some change for transparent material, I haven't carefully look at them yet, but I think this option is needed to export a tailored shader script for transparent material

Reply:
Yup, but for a cycles material to have a transparent "part" it needs to have a shader node that supports it (eg transparent, glass, refraction). By scanning the node tree for these nodes, you can decide if the material is possible to have transparency or not.

Reply:
Oh, I see what you mean,  seems I know quite understand how blender transparent work,  I will look into it